### DISCUSSION TITLE
Use conda env insteade of ubuntu system python3.6

### DISCUSSION THREAD

**Discussion by RangeOfGlitching** [RoboStack/ros-noetic#306] [Q&A] [answered]
hello,
After installing ROS noetic by ROS [doc](http://wiki.ros.org/noetic/Installation/Ubuntu) , I am now able to run CPP and Python programs with ROS. However, some packages do not support Python3.6, so I am wondering if I can use this project to solve my problem.

**Comment by Tobias-Fischer** [comment]
Hi @RangeOfGlitching,

I am not sure if I understand your question correctly. The latest version of the RoboStack builds are based on Python 3.9 - so if you install RoboStack as per our README, you will be able to use ROS with Python 3.9.

We have some really old builds that are based on Python 3.6 but I would not recommend them.

**Comment by RangeOfGlitching** [comment, marked as answer]
Hi @Tobias-Fischer, thanks for your reply. And I'm sorry with my English level(LOL).
When I installed ROS, it will take Ubuntu build-in python as a default interpreter. However I want to use conda to manage the ENV. 
1. Is there any way use conda virture env work with ROS?
2. If I already install ROS, should I uninstall it before conda install ros-noetic?

**Reply by Tobias-Fischer** [reply]
Hi,

1. Yes, this is the goal of RoboStack.
2. There is no need to uninstall, however make sure not to source the system's ROS installation in your bashrc file.

**Reply by RangeOfGlitching** [reply]
Does anything need to be taken into account when using CPP as well?
Thanks

**Reply by Tobias-Fischer** [reply]
See the readme, you need to install the compilers package and a few others.
